lib/Driver/Compilation.cpp

```diff
@@ -291,8 +291,14 @@ namespace driver {
              "not implemented for compilations with multiple jobs");
       if (Comp.ShowJobLifecycle)
         llvm::outs() << "Added to TaskQueue: " << LogJob(Cmd) << "\n";
-      TQ->addTask(Cmd->getExecutable(), Cmd->getArguments(), llvm::None,
-                  (void *)Cmd);
+      if (Cmd->hasResponseFile()) {
```

Comment:
I prefer a style that puts the shorter arm of the if-then-else first, even though it is the do-less arm. I feel that such a choice reduces the max depth of the reader's mental stack. But this is just a suggestion.